test/extended/util/workloads.go

```diff
@@ -0,0 +1,140 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	o "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// Deployment struct defination
+type Deployment struct {
+	name      string
+	namespace string
+	replicas  string
+	appLabel  string
+	image     string
+}
```

Comment:
There is existing `Deployments` lib that defines various functions like `Create` and retrievals of Deployment's fields, for example test/extended/images/resolve.go uses:
`Create`:
```
		deployment, err := oc.KubeClient().AppsV1().Deployments(oc.Namespace()).Create(ctx, &kappsv1.Deployment{
```
Retrievals of Deployment's field:
```
		deployment.Annotations["alpha.image.policy.openshift.io/resolve-names"] = "*"
```
Could we reuse such lib such that avoid dup to redefine?

Reply:
Good suggestion! But if we use this way it will call client-go directly not use our client plugin binary. Maybe it'll miss some check for client plugin binary at e2e scope. I think we should separate reuse for different scenarios. As customer scenario, the customer always use client binary, WDYT?

Reply:
In addition, I think different features needs different deploymets defination origin openshift-tests use testdata yamls to make it simpley for the input parameters of the deployments. I agree we could reuse the exist lib. But from my understanding we should optimize the input of the deployment stuct, for most scenarios we don't need so many attributes. I'll do it later maybe next PR.